module/mempool/stdmap/backend.go

```diff
@@ -213,25 +214,15 @@ func (b *Backend) RegisterEjectionCallbacks(callbacks ...mempool.OnEjection) {
 // reduce will reduce the size of the kept entities until we are within the
 // configured memory pool size limit.
 func (b *Backend) reduce() {
-
 	// we keep reducing the cache size until we are at limit again
-	for len(b.entities) > int(b.limit) {
-
+	// this was a loop, but the loop is now in EjectTrueRandomFast()
```

Comment:
Nit: Is it useful to refer to prior states of the code here?

Reply:
If anyone wants to add a different Eject mechanism, this comment might avoid a new bug.
The change is too small to justify a re-write, so I'm holding my nose and adding comments like this one.